core/src/main/scala/akka/persistence/r2dbc/internal/PubSub.scala

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2021 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.r2dbc.internal
+
+import java.net.URLEncoder
+import java.nio.charset.StandardCharsets
+import java.time.Instant
+import java.util.concurrent.ConcurrentHashMap
+
+import akka.actor.typed.ActorRef
+import akka.actor.typed.ActorSystem
+import akka.actor.typed.Extension
+import akka.actor.typed.ExtensionId
+import akka.actor.typed.pubsub.Topic
+import akka.annotation.InternalApi
+import akka.persistence.Persistence
+import akka.persistence.PersistentRepr
+import akka.persistence.query.TimestampOffset
+import akka.persistence.query.typed.EventEnvelope
+import akka.persistence.typed.PersistenceId
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] object PubSub extends ExtensionId[PubSub] {
+  def createExtension(system: ActorSystem[_]): PubSub = new PubSub(system)
+
+  // Java API
+  def get(system: ActorSystem[_]): PubSub = apply(system)
+
+}
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] class PubSub(system: ActorSystem[_]) extends Extension {
```

Comment:
Looking at this again I wonder if we should perhaps have a small convenience TopicRegistry extension in Akka to make it easy to run one topic actor per topic name in the local actor system rather than having to keep track of those yourself.

Reply:
Created https://github.com/akka/akka/issues/31053 to track that idea on the Akka side.